File: backend/models/product.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Product struct {
	ID          primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	UserID      primitive.ObjectID `bson:"userID" json:"userID"`
	ImageID     primitive.ObjectID `bson:"imageID" json:"imageID"`
	Name        string             `json:"name"`
	Description string             `json:"description"`
	Price       float64            `json:"price"`
	Stock       int                `json:"stock"`
	Category    string             `json:"category"`
	CreatedAt   primitive.DateTime `json:"createdAt"`
	Image       Image              `json:"image,omitempty"`
}

File: frontend/lib/models/product.dart
import './image.dart';

class Product {
  final String id;
  final String name;
  final String description;
  final double price;
  final int stock;
  final String category;
  final String? imageID;
  final Image? image;

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.stock,
    required this.category,
    this.imageID,
    this.image,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: json['price'].toDouble(),
      stock: json['stock'],
      category: json['category'],
      imageID: json['imageID'],
      // imageが存在すればImageオブジェクトを作成
      image: json['image'] != null ? Image.fromJson(json['image']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'price': price,
      'stock': stock,
      'category': category,
      'imageID': imageID,
      'image': image?.toJson(),
    };
  }
}
